Raspberry/lantronixScales/main.js

```diff
@@ -0,0 +1,74 @@
+const net = require('net');
+const fs = require('fs');
+const CameraVapix = require('camstreamerlib/CameraVapix');
+
+let prevWeightData = null;
+let dataBuffer = '';
+
+// Read script configuration
+let settings = null;
+try {
+  let data;
+  if (!process.env.PERSISTENT_DATA_PATH){
+    data = fs.readFileSync('./localdata/settings.json');
+  }else{
+    data = fs.readFileSync(process.env.PERSISTENT_DATA_PATH + 'settings.json');
+  }
+  settings = JSON.parse(data);
+} catch (err) {
+  console.log('No settings file found');
+  return;
+}
+
+// Create camera client for http requests
+let cv = new CameraVapix({
+  'protocol': 'http',
+  'ip': settings.camera_ip,
+  'port': settings.camera_port,
+  'auth': settings.camera_user + ':' + settings.camera_pass,
+});
+
+// Connect to electronic scale
+let client = new net.Socket();
+client.connect(settings.scale_port, settings.scale_ip, () => {
+  console.log('Scale connected');
+  setInterval(() => {
+    client.write(Buffer.from('1B700D0A', 'hex'));
+  }, settings.refresh_rate);
+});
+
+client.on('data', (data) => {
+  dataBuffer += Buffer.from(data, 'hex').toString();
+  const messageEnd = dataBuffer.indexOf('\r\n');
+  if (messageEnd == -1) {
+    return;
+  }
+  const weightData = dataBuffer.substring(0, messageEnd);
+  console.log(weightData);
```

Comment:
Jestli dobre chapu, tak tenhle log se bude vypisovat extreme casto..na jednu stranu chapu jeho dulezitost kdyz neco nefunguje, ale obavam se aby nam nezabil log kamery..?

Reply:
jo je pravda, ze to na kamere nepobezi..promazavame nejak logy na raspberry nebo jen rostou?

Reply:
ne na raspberry to furt jede do nekonečna

Reply:
smazáno